### PR TITLE
don't bail out if a single self-signature fails to verify

### PR DESCRIPTION
**Pull request by maxtaco**
- some keys have bad self-sigs, probably due to old buggy versions of gpg or bad key servers
- can ignore self-signatures that don't verify as long as there is one that does
- make a test case with freacky22527
- add a debugging hash function

### CODE REVIEW

**Comment by maxtaco** [comment]
@patrickxb or @oconnor663, do you mind reviewing?  it turns out this was a pretty easy one to fix...

File: openpgp/packet/public_key.go [hunk 1]
```diff
@@ -664,6 +664,19 @@ func (pk *PublicKey) VerifyRevocationSignature(sig *Signature) (err error) {
 	return pk.VerifySignature(h, sig)
 }
 
+type teeHash struct {
```
Comment (patrickxb) [inline openpgp/packet/public_key.go:667]: Is this used?

Comment (maxtaco) [inline openpgp/packet/public_key.go:667]: used it for debugging. it might be useful in the future

**Comment by patrickxb** [comment]
LGTM

**Pull request merged**